src/foam/nanos/notification/push/PushRegistryService.js

```diff
@@ -12,25 +12,47 @@ foam.CLASS({
   javaImports: [
     'foam.dao.DAO',
     'foam.nanos.auth.Subject',
-    'foam.nanos.auth.User'
+    'foam.nanos.auth.User',
+    'foam.nanos.auth.AuthService',
+    'foam.nanos.notification.push.iOSNativePushRegistration'
   ],
 
   methods: [
     {
       name: 'subscribe',
 //      type: 'Void',
-//      args: 'Context x, String sub, String endpoint, String key, String auth'
+//      args: 'Context x, String sub, String endpoint, String key, String auth, String token'
       javaCode: `
         User user = ((Subject) x.get("subject")).getUser();
 
         if ( user == null ) throw new IllegalArgumentException("Missing user.");
+        PushRegistration r;
+        if ( token == null ) {
+          r = new PushRegistration();
+          r.setEndpoint(endpoint);
+          r.setKey(key);
+          r.setAuth(auth);
+        } else {
+          iOSNativePushRegistration ir =  (iOSNativePushRegistration) new iOSNativePushRegistration();
+          ir.setEndpoint(token);
```

Comment:
token is null so will the endpoint, is null end point expected?

Reply:
This is the else block